### PR TITLE
Feature/logical operators

### PR DESCRIPTION
**Pull request by SimonSchick**
Things I've done:
 - Refactor:
   - Split request manager tests into separate file (as it should have been from the start 😛 )
   - Split query stringification logic into own class (same for tests)
   - Added logical operator support
     - This is heavily based on type checking.

I am unsure on how to deal with conjunction here.

### CODE REVIEW

File: src/Types.ts [hunk 1]
```diff
@@ -28,6 +28,25 @@ export type Comparator = PropertyType | {
...
+export type Conjunction = {
+    $and: ConjunctionArgument;
+}
+
+export type DisjunctionArgument = {
+    [prop: string]: Comparator | Conjunction;
+};
+export type Disjunction = {
+    $or: DisjunctionArgument;
+}
```
Comment (connor4312) [inline src/Types.ts]: Prefer interfaces over `type` defs when possible, they're a bit easier to deal with and are more flexible

File: src/adapters/components/QueryBuilder.ts [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+            return String(Math.floor(value.getTime() / 1000));
+        }
+        if (typeof value === 'boolean') {
+            return value ? '1' : '0';
+        }
+
+        return String(value).replace(/[;:,]/g, '\\$&');
+    }
+
+    public conditionToString (condition: { [s: string]: WhereClause } ): string {
```
Comment (connor4312) [inline src/adapters/components/QueryBuilder.ts]: This typings isn't reflected in the QueryOptions

File: src/adapters/components/QueryBuilder.ts [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+
+                out.push(`${key}:eq:${this.escapeComparatorValues(comparator)}`);
+                return;
+            }
+
+            if (isDisjunction(comparator, key)) {
+                out.push(comparator.map(v => `(${this.conditionToString(v)})`).join(':or:'));
+                return;
+            }
+
```
Comment (connor4312) [inline src/adapters/components/QueryBuilder.ts]: Fyi conjunctions are equivalent to the comma-separated clauses we have previously, they can be compiled down if you want

Comment (SimonSchick) [inline src/adapters/components/QueryBuilder.ts]: You mean flattened?

Comment (connor4312) [inline src/adapters/components/QueryBuilder.ts]: Yes. E.g.

```
a:eq:b,b:eq:c \equiv (a.eq.b):and:(b:eq:c)
```

Comment (SimonSchick) [inline src/adapters/components/QueryBuilder.ts]: We decided to drop conjunctions for now until they become a use case.

File: src/adapters/components/QueryBuilder.ts (hunk 2 above)
Comment (SimonSchick) [inline src/adapters/components/QueryBuilder.ts]: It is now.

**Pull request merged**